crates/uv/tests/it/pip_compile_scenarios.rs

```diff
@@ -422,6 +422,17 @@ fn python_patch_override_patch_compatible() -> Result<()> {
     let requirements_in = context.temp_dir.child("requirements.in");
     requirements_in.write_str("python-patch-override-patch-compatible-a==1.0.0")?;
 
+    uv_snapshot!(filters, context.venv().arg("-p").arg("3.9.20"), @r"
```

Comment:
Why is this here? We can't mutate the scenario files, they're generated.

Reply:
Sorry, I thought I commented on this. This is here to prevent the test from using the highest available patch version if a minor link directory has been installed. Our test environments seem like they're not isolated enough to be able to simulate only having certain patches installed.

Reply:
I think that means we might want to canonicalize Python executables before using them in the test suite? There are various tests that do require specific patch versions, and rely on that `TestContext` behavior.

Reply:
Maybe we can canonicalize them only with the `python-patch` feature

Reply:
Updated in #14029